CONTRIBUTING.rst

```diff
@@ -162,3 +162,150 @@ their individual contributions.
+reviewer to try to ensure that the answer to both of those questions
+is no.
+
+Ideally of course the change should also make one or both of the users'
+and our lives *better*, but it's OK for changes to be mostly neutral.
+The author should be presumed to have a good reason for submitting the
+change in the first place, so neutral is good enough!
+
+The rest of this section outlines specific things reviewers should
+focus on in aid of this, broken up by sections according to their area
+of applicability.
+
+All of these conditions must be satisfied for merge. Where the reviewer
+thinks this conflicts with the above higher level goals, they may make
+an exception if both the author and another maintainer agree.
+
+~~~~~~~~~~~~~~~~~
+General Questions
+~~~~~~~~~~~~~~~~~
+
+The following are required for almost every change:
+
+1. Changes must be of reasonable size. If a change could logically
+   be broken up into several smaller changes that could be reviewed
+   separately on their own merits, it should be.
+2. The motivation for each change should be clearly explained (this
+   doesn't have to be an essay, especially for small changes, but
+   at least a sentence of explanation is usually required).
+3. The likely consequences of a change should be outlined (again,
+   this doesn't have an essay, and it may be sufficiently
+   self-explanatory that the motivation section is sufficient).
+
+~~~~~~~~~~~~~~~~~~~~~
+Functionality Changes
+~~~~~~~~~~~~~~~~~~~~~
+
+This section applies to any changes in Hypothesis's behaviour, regardless
+of their nature.
+
+1. The code should be clear in its intent and behaviour.
+2. Behaviour changes should come with appropriate tests to demonstrate
+   the new behaviour.
+3. Hypothesis must never be *flaky*. Flakiness here is
+   defined as anything where a test fails and this does not indicate
+   a bug in Hypothesis or in the way the user wrote the code or the test.
+
+~~~~~~~~~~~
+API Changes
+~~~~~~~~~~~
+
+Public API changes require the most careful scrutiny of all reviews,
+because they are the ones we are stuck with for the longest: Hypothesis
+follows semver and additionally tries not to break API compatibility
+wherever possible.
+
+When a public API changes we should ask the following questions:
+
+1. All public API changes must be well documented. If it's not documented,
+   it doesn't count as public API!
+2. Changes must be backwards compatible. Where this is not possible, they
+   must first introduce a deprecation warning, then once the major version
+   is bumped the deprecation warning and the functionality may be removed.
+3. If an API is deprecated, the deprecation warning must make it clear
+   how the user should modify their code to adapt to this change (
+   possibly by referring to documentation).
+4. If it is likely that we will want to make backwards compatible changes
+   to an API later, to whatever extent possible these should be made immediately
+   when it is introduced instead.
```

Comment:
Surely it’s more important to do this for backwards incompatible changes? If a future change is backwards compatible, doing it later doesn’t matter.